Tests/FluentAssertions.Specs/Primitives/ReferenceTypeAssertionsSpecs.Satisfy.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Primitives;
+
+public partial class ReferenceTypeAssertionsSpecs
+{
+    [Fact]
+    public void When_object_satisfies_inspector_it_should_not_throw()
+    {
+        // Arrange
+        var someObject = new object();
+
+        // Act
+        Action act = () => someObject.Should().Satisfy<object>(x => x.Should().NotBeNull());
```

Comment:
We usually do not write this `Action x....; x.Should.NotThrow()` anymore, because if it fails, it fails anyway

Reply:
Okay?
It just combines the act and assert, which I find a bit awkward. The reason it fails is also unclear from the test since it might fail for any number of reasons.
However, I'm fine doing whatever you find best and there is no consistency in how the tests are written, so it's difficult to get right. I tried to follow the pattern of the tests for the class I am modifying, but even in that class, there is no consistency.

@dennisdoomen Is there a preferred way of writing tests? I can't find any guidelines, so I'd like some "official" answer to this.
The same goes for the usage of nested classes, although I can that pattern being used in several places, I just missed that.

Reply:
For the test that are not supposed to throw, we do not use the `NotThrow` anymore.

Reply:
I've merged the Act / Assert for non-throwing assertions